analyzer/src/main/kotlin/curation/FilePackageCurationProvider.kt

```diff
@@ -28,11 +28,13 @@ import org.ossreviewtoolkit.model.PackageCuration
 import org.ossreviewtoolkit.model.readValueOrDefault
 
 /**
- * A [PackageCurationProvider] that loads [PackageCuration]s from a single file. Supports all file formats specified
- * in [FileFormat].
+ * A [PackageCurationProvider] that loads [PackageCuration]s from all [curationFiles]. Supports all file formats
```

Comment:
Take a look at `(Simple)PackageConfigurationProvider`, I find it quite nice that is has `configurations: Collection<PackageConfiguration>` as opposed to `List<File>`. I believe at least considering aligning with that approach would make sense.

Reply:
While I agree that in general it's better to handle defined objects, for the `FilePackageCurationProvider` I think it's better to pass the actual file.

Reply:
Actually, I'd agree with @fviernau here. If we go into the direction of an ORT "server" backend and a frontend, then our APIs should abstract away from `File`s where possible. I argumented similarly in e.g. [this change](https://github.com/oss-review-toolkit/ort/pull/3926/commits/6d048c02236d4129584bc893b3fc6b4bf71f2f36).

However, such a refactoring should be done in a separate commit, or maybe even in a preparing separate PR, IMO. Esp. as probably more changes make sense to do then, like renaming "*File*PackageCurationProvider" to "*Simple*PackageCurationProvider".

That said, such a refactoring could also be done *after* merging this PR.

Reply:
Agree, out of scope of this one makes sense. Just a small hint: testing is also easier without files.